src/Symfony/Component/Image/Tests/Constraint/IsImageEqual.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Image\Tests\Constraint;
+
+use PHPUnit\Framework\Constraint\Constraint;
+use Symfony\Component\Image\Image\ImageInterface;
+use Symfony\Component\Image\Image\Histogram\Bucket;
+use Symfony\Component\Image\Image\Histogram\Range;
+
+if (class_exists(\PHPUnit_Framework_Constraint::class)) {
+    abstract class PHPUnitConstraint extends \PHPUnit_Framework_Constraint{}
+} else {
+    abstract class PHPUnitConstraint extends Constraint{}
+}
+
+class IsImageEqual extends PHPUnitConstraint
+{
+    /**
+     * @var \Symfony\Component\Image\Image\ImageInterface
+     */
+    private $value;
+
+    /**
+     * @var float
+     */
+    private $delta;
+
+    /**
+     * @var integer
+     */
+    private $buckets;
+
+    /**
+     * @param \Symfony\Component\Image\Image\ImageInterface $value
+     * @param float                        $delta
+     * @param integer                      $buckets
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct($value, $delta = 0.1, $buckets = 4)
+    {
+        if (!$value instanceof ImageInterface) {
+            throw \PHPUnit_Util_InvalidArgumentHelper::factory(1, ImageInterface::class);
+        }
+
+        if (!is_numeric($delta)) {
+            throw \PHPUnit_Util_InvalidArgumentHelper::factory(2, 'numeric');
```

Comment:
this won't work on PHPUnit 6, as the non-namespaced class does not exist anymore